executor/src/witgen/jit/witgen_inference.rs

```diff
@@ -76,15 +97,64 @@ impl<'a, T: FieldElement, FixedEval: FixedEvaluator<T>> WitgenInference<'a, T, F
             }
             Identity::Connect(_) => ProcessResult::empty(),
         };
-        self.ingest_effects(result.effects);
-        result.complete
+        self.ingest_effects(result)
+    }
+
+    pub fn process_input(
```

Comment:
I'm wondering if it's nice to combine `process_input`, `process_output` and `set` by pulling `from_known_symbol`/`from_unknown_symbol` out to the caller?

The reason is that these two functions introduce the concept of input and output which I think was not relevant for this module until this point (while known and unknown symbols are already a concept known to the caller due to the constructor argument "known_variables").